internal/util/askserver.go

```diff
@@ -0,0 +1,63 @@
+package util
+
+import (
+	"context"
+	"slices"
+
+	"github.com/10gen/migration-verifier/mslices"
+	"github.com/10gen/migration-verifier/option"
+	"github.com/pkg/errors"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+// ServerThinksTheseMatch runs an aggregation on the server that determines
+// whether the server thinks a & b are equal. This allows you, e.g., to
+// ignore BSON type differences for equivalent numbers.
+//
+// tinker is an optional pipeline that operates on the documents in the
+// pipeline (`a` and `b`, respectively) before they're compared.
+func ServerThinksTheseMatch(
+	ctx context.Context,
+	client *mongo.Client,
```

Comment:
I actually think it’s better as a static function so that there’s better separation of concerns. The more things we “hang” on Verifier, the less modular it all is.

Reply:
Can we add a comment that to the function that explains it expects the client to only be the meta client for most of the cases?

Reply:
done